pkg/kubectl-argo-rollouts/cmd/lint/lint.go

```diff
@@ -261,6 +261,11 @@ func setIngressManagedAnnotation(rollouts []v1alpha1.Rollout, refResource valida
 	for _, rollout := range rollouts {
 		for i := range refResource.Ingresses {
 			var serviceName string
+
+			// Basic Canary so ingress is only pointing a single service and so no linting is needed for this case.
+			if rollout.Spec.Strategy.Canary == nil || rollout.Spec.Strategy.Canary.TrafficRouting == nil {
```

Comment:
The `valid-nginx-basic-canary.yml` manifest you added has no `Spec.Strategy.Canary.TrafficRouting` field, which tests the second part of this clause, but it does have a `Spec.Strategy.Canary` field.

Should we add another test Rollout manifest with no `Spec.Strategy.Canary` field to test the former part of this clause? Apologies if I'm misunderstanding the original bug

Reply:
Yea the bug was really only ever for the missing TrafficRoutering field for basic canary there is already test cases such as blue green https://github.com/argoproj/argo-rollouts/blob/754a5c7d0f1e7d4a26f26f8ff594a64dede82886/pkg/kubectl-argo-rollouts/cmd/lint/testdata/valid-blue-green.yml where we do not check for referenced resources.

Reply:
Thanks for clarifying!